src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -853,7 +853,7 @@ public static IPredictorProducing<Float> CreateCalibratedPredictor(IHostEnvironm
             var predWithFeatureScores = predictor as IPredictorWithFeatureWeights<Float>;
             if (predWithFeatureScores != null && predictor is IParameterMixer<Float> && cali is IParameterMixer)
                 return new ParameterMixingCalibratedPredictor(env, predWithFeatureScores, cali);
-            if (needValueMapper)
+            if (needValueMapper || predictor is IValueMapper)
                 return new CalibratedPredictor(env, predictor, cali);
             return new SchemaBindableCalibratedPredictor(env, predictor, cali);
```

Comment:
Hypothetically, if `needValueMapper` were true, but `predictor` was *not* an `IValueMapper`, what would happen? Would it throw when we attempted to instantiate `CalibratedPredictor`?

The comments for this indicate that this is needed by OVA. If it is only useful there, I wonder if we could handle the error condition of the predictor *not* being an `IValueMapper` there since in that case it would be possible to provide reasonably friendly, actionable error message. Then we could get rid of the parameter `needValueMapper` within this file.